source/common/network/utility.cc

```diff
@@ -50,29 +53,33 @@ Address::InstanceConstSharedPtr Utility::resolveUrl(const std::string& url) {
 }
 
 bool Utility::urlIsTcpScheme(const std::string& url) { return url.find(TCP_SCHEME) == 0; }
-
+bool Utility::urlIsUdpScheme(const std::string& url) { return url.find(UDP_SCHEME) == 0; }
```

Comment:
huh, I'd expect this to be absl::StartsWith but I suspect we're better off with consistency so worth leaving as-is.
Just for curiosity's sake, @mattklein123 do we have use cases where we expect the tcp:// to not be at the beginning?  Did config have whitespace preface at some point?

Reply:
It's only prefix. I think this was so early on there probably wasn't even `StringUtil::startsWith` back then. Feel free to change this to absl::StartsWith. You could also change these functions to take a string_view also if you like.

Reply:
Ok, let's make UDP startsWith and add a TODO either for @conqerAtapple  or for me to update the other two?
After the joys of the xff whitespace change I prefer making "this should be fine" changes to existing code in standalone PRs :-P

Reply:
This is already checking that it's at position 0, so I think it can be changed without any fear? It's just a perf optimization.

Reply:
Oh yeah, totally blanked on that.  Yeah, let's change it!

Reply:
@alyssawilk I assume it is case sensitive.

Reply:
Yes